doc/psud/PSU_daemon_design.md

```diff
@@ -88,14 +157,41 @@ Commands:
 
 The current output for "show platform psustatus" looks like:
 
+```
+admin@sonic:~$ show platform  psustatus
+PSU    Model          Serial        HW Rev      Voltage (V)    Current (A)    Power (W)  Status  LED
+-----  -------------  ------------  --------  -------------  -------------  -----------  ------- -----
+PSU 1  MTEF-PSF-AC-A  MT1629X14911  A3                12.08           5.19        62.62  WARNING green
+PSU 2  MTEF-PSF-AC-A  MT1629X14913  A3                12.01           4.38        52.50  OK      green
+```
+
+The field `Status` represents the status of the PSU, which can be the following:
+- `OK` which represents no alarm raised due to PSU power exceeding the threshold
+- `Not OK` which can be caused by:
+  - power is not good, which means the PSU is present but no power
+- `WARNING` which can be caused by:
+  - power exceeds the PSU's power threshold
+
+### 4.2 psuutil
+
+`psuutil` fetches the information via calling platform API directly. Both warning and critical thresholds will be exposed in the output of psuutil status.
+The "WARNING" state is not exposed because psuutil is a one-time command instead of a daemon, which means it does not store state information. It fetches information via calling platform API so it can not distinguish the following status:
+
+1. The power exceeded the critical threshold but is in the range between the warning and critical thresholds, which means the alarm should be raised
+2. The power didn't exceed the critical threshold and exceeds the warning threshold, which means the alarm should not be raised
+
+An example of output
 ```
 admin@sonic:~$ show platform psustatus
-PSU    Model          Serial        HW Rev      Voltage (V)    Current (A)    Power (W)  Status    LED
------  -------------  ------------  --------  -------------  -------------  -----------  --------  -----
-PSU 1  MTEF-PSF-AC-A  MT1629X14911  A3                12.09           5.44        64.88  OK        green
-PSU 2  MTEF-PSF-AC-A  MT1629X14913  A3                12.02           4.69        56.25  OK        green
+PSU   Model         Serial       HW Rev   Voltage (V)   Current (A)   Power (W)   Power Warn Thres (W)   Power Crit Thres (W)    Status LED
+----- ------------- ------------ -------- ------------- ------------- ----------- ---------------------- ---------------------- ------- -----
+PSU 1 MTEF-PSF-AC-A MT1843K17965 A4               12.02          3.62       43.56                  38.00                  58.00 OK      green
+PSU 2 MTEF-PSF-AC-A MT1843K17966 A4               12.04          4.25       51.12                  38.00                  58.00 OK      green
+
 ```
 
+In case neither threshold is supported on the platform, `N/A` will be displayed.
+
 ## 5. PSU LED management
 
 The purpose of PSU LED management is to notify user about PSU event by PSU LED or syslog. Current PSU daemon psud need to monitor PSU event (PSU voltage out of range, PSU too hot) and trigger proper actions if necessary.
```

Comment:
In case of warning status do we need also to set the led?

Reply:
no. we will change the system status led.